src/components/sideBarContentEdit/SelectField.tsx

```diff
@@ -293,3 +305,85 @@ export const SingleSelectField = ({
     </Select>
   );
 };
+
+// function to return dir based on the currentPath
+
+function findAndReturnDirs(
+  objArray: IDir[],
+  path: string[],
+  depth: number,
+  index?: number
+): IDir[] {
+  if (depth === 0) {
+    return []; // Return null if the desired depth is reached, but the path is not found.
+  }
+
+  if (index === 1) {
+    return objArray;
+  }
+
+  for (const obj of objArray) {
+    if (obj.slug === path[depth - 2]) {
+      if (path.length === 2 && depth === 2) {
+        return obj.nestDir || []; // Return the node if the path is found at the desired depth.
+      } else if (obj.nestDir) {
+        const subdirectory: IDir[] = findAndReturnDirs(
+          obj.nestDir,
+          path.slice(1),
+          depth - 1
+        );
+        if (subdirectory !== null) {
+          return subdirectory || []; // Return the subdirectory if found at the desired depth.
+        }
+        return objArray;
+      }
+    }
+  }
+
+  return []; // Return empty if the path is not found.
+}
+
+// To enable users input a custom directory
+export const OnlySelectDirectory = ({
+  name,
+  editedData,
+  updateData,
+  index,
+  path,
+  autoCompleteList,
+  userCanAddToList,
+}: PropsDirectory) => {
+  const handleAddItem = (value: string) => {
+    updateData(value);
+  };
+  const directoriesInPath = findAndReturnDirs(
+    autoCompleteList,
+    path.split("/"),
+    index + 1,
+    index + 1
+  );
+  const [value, setValue] = useState<string>(editedData);
+  const handleAutoCompleteSelect = (data: AutoCompleteData) => {
+    handleAddItem(data.value);
+    setValue(data.value);
+  };
+
+  // remove previously selected option from list
+  const newAutoCompleteList =
```

Comment:
This is for selecting multiple items. Selecting location is single use so there's no need to remove previous items from the list